core/src/main/scala/kafka/server/ReplicaFetcherThread.scala

```diff
@@ -115,7 +115,12 @@ class ReplicaFetcherThread(name: String,
   override def initiateShutdown(): Boolean = {
     val justShutdown = super.initiateShutdown()
     if (justShutdown) {
-      leaderEndpoint.close()
+      try {
+        leaderEndpoint.close()
+      } catch {
+        case t: Throwable =>
+          error(s"Fail to close leader endpoint $leaderEndpoint after initiating replica fetcher thread shutdown", t)
```

Comment:
Since it is OK for `leaderEndPoint.close()` to throw exception and it is actually expected in operation, would it be reasonable to log this at debug level and add Java comment to explain why it is OK and expected?

Reply:
Sure. Done.